src/finch/symbolic/term.py

```diff
@@ -71,6 +72,30 @@ def children(self) -> list[Term]:
         ...
 
 
+@dataclass(frozen=True, eq=True)
+class LiteralRepr:
```

Comment:
Is `LeafRepr` better? Or `EvaluableRepr`?

Reply:
I think it would be better as a function than a base class, right? Since we may want to use this logic on formats, or fields other than val sometimes, for example.

Reply:
Can you give an example of Field or Format that wouldn't fit here?

Reply:
I think you inferred that I was asking for a bigger change than I was. What I mean is more that I'd like the logic of LiteralRepr to be put into a function `literal_repr`, and classes which inherit from LiteralRepr to instead define 
```
    def __repr__(self):
        return literal_repr(("val", self.val))
```

The reason I thought to push for this was that automatically traversing `__dir__` fields feels hacky.

Reply:
Ah, right. Done! I changed the `LiteralRepr` class to a function.